spec/components/layout_for_public_spec.rb

```diff
@@ -96,4 +96,59 @@ def component_name
 
     assert_select ".govuk-header__link--homepage[href='https://example.com/jam']"
   end
+
+  it "contains real user metrics loader script" do
+    visit "/public"
+    assert page.has_selector?("html > head > script[src*='rum-loader']", visible: :all)
+  end
+
+  it "does not contain real user metrics scripts before cookie banner interacted with" do
+    Capybara.current_driver = Capybara.javascript_driver
+
+    visit "/public"
+
+    assert page.has_selector?("html > head > script[src*='rum-loader']", visible: :all)
+    assert page.has_no_selector?("html > head > script[src*='lux/lux']", visible: :all)
+    assert page.has_no_selector?("html > head > script[src*='lux/lux-polyfill']", visible: :all)
+  end
+
+  it "does not contain real user metrics scripts on the page where cookies are accepted" do
```

Comment:
Looks like this is testing that the scripts aren't immediately loaded when the user accepts cookies. Just wanted to highlight some recent changes to the gem that allow for JS like this to be executed as soon as cookies are accepted, if that could be useful here: https://github.com/alphagov/govuk_publishing_components/blob/master/docs/javascript-modules.md#modules-and-cookie-consent

Reply:
This is on purpose - I couldn't tell what would happen to the metrics if a user accepts cookies a while after loading a page. I then opted to only run this on the page visited after a user has accepted cookies rather than possibly get spurious data.

Reply:
Okay, that sounds like a sensible approach.